src/libsystemd/sd-bus/bus-internal.c

```diff
@@ -91,8 +92,17 @@ bool interface_name_is_valid(const char *p) {
                                 (!dot && *q >= '0' && *q <= '9') ||
                                 *q == '_';
 
-                        if (!good)
+                        if (!good) {
+                                if (DEBUG_LOGGING) {
+                                        _cleanup_free_ char *iface = NULL;
+                                        iface = cescape(p);
+                                        if (!iface)
+                                                log_oom();
+                                        else
+                                                log_debug("The interface %s is invalid as it contains special character %c", iface, *q);
```

Comment:
Imagine that for example the offending character is `\n`. You can't just print it, for the exact same reason you can't print the whole string with the character. I'd suggest the following:
`log_debug("Interface '%s' is invalid as it contains a special character.", strnull(iface));`